tests/exclude_filter/test

```diff
@@ -0,0 +1,146 @@
+ok($result, 0); #3
+system("auditctl -d exclude,always -F uid=$uid");
+$result = system("auditctl -a exclude,always -F gid=$gid");
+ok($result, 0); #4
+system("auditctl -d exclude,always -F gid=$gid");
+$result = system("auditctl -a exclude,always -F auid=$auid");
+ok($result, 0); #5
+system("auditctl -d exclude,always -F auid=$auid");
+$result = system("auditctl -a exclude,always -F loginuid_set=0");
+ok($result, 0); #6
+system("auditctl -d exclude,always -F loginuid_set=0");
+$result = system("auditctl -a exclude,always -F subj_user=$subj_user");
+ok($result, 0); #7
+system("auditctl -d exclude,always -F subj_user=$subj_user");
+$result = system("auditctl -a exclude,always -F subj_role=$subj_role");
+ok($result, 0); #8
+system("auditctl -d exclude,always -F subj_role=$subj_role");
+$result = system("auditctl -a exclude,always -F subj_type=$subj_type");
+ok($result, 0); #9
+system("auditctl -d exclude,always -F subj_type=$subj_type");
+$result = system("auditctl -a exclude,always -F subj_sen=$subj_sen");
+ok($result, 0); #10
+system("auditctl -d exclude,always -F subj_sen=$subj_sen");
+$result = system("auditctl -a exclude,always -F subj_clr=$subj_clr");
+ok($result, 0); #11
+system("auditctl -d exclude,always -F subj_clr=$subj_clr");
+
+$result = system("auditctl -a exclude,always -F ppid=$ppid >/dev/null 2>&1");
+ok($result ne 0); #12
+system("auditctl -d exclude,always -F ppid=$ppid >/dev/null 2>&1");
+$result = system("auditctl -a exclude,always -F euid=$euid >/dev/null 2>&1");
+ok($result ne 0); #13
+system("auditctl -d exclude,always -F euid=$euid >/dev/null 2>&1");
+$result = system("auditctl -a exclude,always -F obj_user=$obj_user >/dev/null 2>&1");
+ok($result ne 0); #14
+system("auditctl -d exclude,always -F obj_user=$obj_user >/dev/null 2>&1");
+
+$result = system("auditctl -a exclude,always -F msgtype=$msgtype -F pid=$pid -F uid=$uid -F gid=$gid -F auid=$auid -F subj_user=$subj_user -F subj_role=$subj_role -F subj_type=$subj_type -F subj_sen=$subj_sen -F subj_clr=$subj_clr");
+ok($result, 0); #15
+
+$result = system("auditctl -a exit,always -F arch=b64 -S all -F path=/tmp/$key");
+ok($result, 0); #16
+usleep 100000;
+
+open(my $tmpfile, ">", "/tmp/$key");
+#write($tmpfile, "\n");
+close($tmpfile);
+usleep 100000;
+
+# test for the SYSCALL message provoked by creat
+my $result = system("ausearch -i -m SYSCALL -p $pid -ui $uid -gi $gid -ul $auid -su $subj_user:$subj_role:$subj_type:$subj_sen-$subj_clr -ts recent > $stdout 2> /dev/null"); #$stderr");
+ok($result, 256); #17
+
+my $found_msg = 0;
+my $line;
+while ($line = <$fh_out>) {
+	$found_msg = 1;
+}
+ok($found_msg, 0); #18
+
+$result = system("auditctl -d exclude,always -F msgtype=$msgtype -F pid=$pid -F uid=$uid -F gid=$gid -F auid=$auid -F subj_user=$subj_user -F subj_role=$subj_role -F subj_type=$subj_type -F subj_sen=$subj_sen -F subj_clr=$subj_clr");
+ok($result, 0); #19
+usleep 100000;
+
+unlink "/tmp/$key";
+usleep 100000;
```

Comment:
Hmm, I hate magic sleep values like this, although I recognize that we may have little choice here.  Is it worth forcing a FS sync as well before the sleep?